kyaml/yaml/match_test.go

```diff
@@ -61,7 +61,7 @@ spec:
 			"- nginx:1.7.9\n"},
 		{[]string{
 			"spec", "template", "spec", "containers", "[name=.*]", "ports"},
-			"- - containerPort: 80\n- - containerPort: 8081\n  - containerPort: 9090\n"},
+			"-\n- containerPort: 80\n-\n- containerPort: 8081\n- containerPort: 9090\n"},
```

Comment:
@mikebz @KnVerey this one is the most concerning to me:

```
-
- containerPort: 80
-
- containerPort: 8081
- containerPort: 9090
```

Reply:
Agreed, those two do not look semantically equivalent. Before is like `[[80], [8081, 9090]]` after is like `[nil, 80, nil, 8081, 9090]`. Can you confirm whether this was introduced by Mike's PR or is already a problem at the head of v3 branch?

Reply:
It looks like an issue with Mike's PR. I'll investigate it tomorrow to see if we can fix it.

Reply:
sorry a little late to the party, I will take a look as well.

Reply:
As an update, I did some digging and it seems like a tricky thing to fix - might take more than a day. I thought I had a fix but it broke another test. I'll keep at it.

Reply:
@KnVerey It looks like [Donny's PR](https://github.com/Shell32-Natsu/yaml/commit/dd16c726352a7e6d641105100ec056a9f92fb7fd) solves the problem of nested lists, we can use what we need to from his code.

Reply:
Resolved with go-yaml/yaml#753